ReconnaissanceTools/ShodanAPI.py

```diff
@@ -0,0 +1,111 @@
+import os
+import shodan
+import sys
+import tkinter as tk
+from nav_bar import *
+
+
+class ShodanScript(tk.Frame):
+    def __init__(self, parent, controller):
+        tk.Frame.__init__(self, parent)
+        self.controller = controller
+        display_nav_bar(self, controller)
+        
+        shodanframe = tk.Label(self, text="Shodan Reconnaissance", bg='#3B5262', fg='white', anchor="c")
+        shodanframe.place(rely=0.08, relheight=0.12, relwidth=1)
+
+        allscreenframe = tk.Label(self, bg='white')
+        allscreenframe.place(rely=0.2, relheight=1, relwidth=1)
+
+
+        Label(self, text="Enter Target IP Address:", bg='#4D6C84', fg='white', anchor='c').place(rely=0.38, relx=0.08, relheight=0.08, relwidth=0.24)
+        
+        ##entry bar
+        self.entry = Entry(self)
+        self.entry.place(rely=0.38, relx=0.32, relheight=0.08, relwidth=0.3)
+        self.pageNumber = 1
+
+        ##default labels for the grid
+        a=b=c=d=e=f=g=h=i=j=k=l=m=n=o=p=q=r = Label(self, bg='#4D6C84', fg='white', anchor='c')
+        self.LabelsList = [a,b,c,d,e,f,g,h,i,j,k,k,l,m,n,o,p,q,r]
+
+        ##buttons
+        Button(self, text="Start", bg='#4D6C84', fg='white', anchor='c', command=self.ShodanMain).place(rely=0.38, relx=0.66, relheight=0.08, relwidth=0.08)
+        self.nextButton = Button(self, text="Next", bg='#4D6C84', fg='white', anchor='c')
+        self.prevButton = Button(self, text="Previous", bg='#4D6C84', fg='white', anchor='c')
+
+
+
+    def ShodanMain(self):
+
+        # API Key needs to be added to the Environment Variable prior to use.
+        api_key = ['vvxN0nHpOrATgGiP4nimlQGSiNzb00eo'] #made specifically for this project
```

Comment:
Even if for this project, probably not a great practice to have an API key visible in our source code